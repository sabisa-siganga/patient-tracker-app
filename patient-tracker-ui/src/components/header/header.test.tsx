import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import Header from "./header";
import { BrowserRouter } from "react-router-dom";

describe("Renders: <Header />", () => {
  it("renders the header with the correct user", () => {
    const currentUser = "John Doe";

    render(
      <BrowserRouter>
        <Header currentUser={currentUser} />
      </BrowserRouter>
    );

    // Check if the header text is rendered
    expect(screen.getByText(currentUser)).toBeInTheDocument();

    // Check if the SignoutBtn component is rendered
    expect(
      screen.getByRole("button", { name: /Sign Out/i })
    ).toBeInTheDocument();
  });

  it("Taking a snapshot", () => {
    const currentUser = "John Doe";

    const tree = renderer
      .create(
        <BrowserRouter>
          <Header currentUser={currentUser} />
        </BrowserRouter>
      )
      .toJSON();

    // Check if the snapshot matches the rendered component
    expect(tree).toMatchSnapshot();
  });
});
