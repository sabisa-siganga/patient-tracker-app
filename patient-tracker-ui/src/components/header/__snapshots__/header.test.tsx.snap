// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders: <Header /> Taking a snapshot 1`] = `
<div
  className="header-container mb-4"
>
  <p>
    John Doe
  </p>
  <div
    className="signout-container"
  >
    <button
      className="btn btn-dark"
      onClick={[Function]}
    >
      Sign out
    </button>
  </div>
</div>
`;
